hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/SourceFormatAdapter.java

```diff
@@ -92,6 +96,9 @@ public SourceFormatAdapter(Source source) {
         return ((RowSource) source).fetchNext(lastCkptStr, sourceLimit);
       case AVRO: {
         InputBatch<JavaRDD<GenericRecord>> r = ((AvroSource) source).fetchNext(lastCkptStr, sourceLimit);
+        if (r.getSchemaProvider() == null) {
```

Comment:
should we just have `getSchemaProvider()` throw this as a HoodieRuntimeException?

Reply:
makes sense! That makes the code much cleaner. Thank you for pointing this out.